[3] move aten targets to shared buck file

linbinyu opened the pull request:
Summary: move aten related targets to the shared buck file

Test Plan: Github CI, sandcastle

Differential Revision: D37323401

iseeyuan reviewed (approved):
LGTM. Please make sure the internal CI passes.

linbinyu committed:
[3] move aten and code gen to shared buck file (#79966)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/79966

move aten related targets to the shared buck file.

Leaving pt_operater_library intact for D36912042

Test Plan: Github CI, sandcastle

Reviewed By: iseeyuan

Differential Revision: D37323401

fbshipit-source-id: a58f87bd77ea642d1c2cb0bc2411e79d2338c2ef
